<a href="https://colab.research.google.com/github/LimYeongJun/Jump-to-python/blob/master/Jump_to_python_%EC%97%B0%EC%8A%B5%EB%AC%B8%EC%A0%9C_2%EC%9E%A5_%26_3%EC%9E%A5_Copy1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 02장 연습문제

**Q1. 홍길동 씨의 과목별 점수는 다음과 같다. 홍길동 씨의 평균 점수를 구해 보자.**

In [0]:
a = 80
b = 75
c = 55

(a+b+c) / 3

70.0

**Q2. 자연수 13이 홀수인지 짝수인지 판별할 수 있는 방법에 대해 말해 보자.**

In [0]:
13 % 2

1

위 결과와 같이 나머지가 1로 출력되므로 **홀수** 라고 할 수 있다.

**Q3. 홍길동 씨의 주민등록번호는 881120-1068234이다. 홍길동 씨의 주민등록번호를 연월일(YYYYMMDD) 부분과 그 뒤의 숫자 부분으로 나누어 출력해 보자.**

In [0]:
a = "881120-1068234"
yyyymmdd = a[:6]
num = a[7:]
print(yyyymmdd)
print(num)

881120
1068234


**Q4. 주민등록번호 뒷자리의 맨 첫 번째 숫자는 성별을 나타낸다. 주민등록번호에서 성별을 나타내는 숫자를 출력해 보자.**

In [0]:
pin = "881120-1068234"
pin[7]

'1'

**Q5. 다음과 같은 문자열 a:b:c:d가 있다. 문자열의 replace 함수를 사용하여 a#b#c#d로 바꿔서 출력해 보자.**

In [0]:
a = "a:b:c:d"
b = a.replace(":", "#")
print(b)

a#b#c#d


**Q6. [1, 3, 5, 4, 2] 리스트를 [5, 4, 3, 2, 1]로 만들어 보자.**

In [0]:
a = [1,3,5,4,2]
a.sort()
a.reverse()
print(a)

[5, 4, 3, 2, 1]


**Q7. ['Life', 'is', 'too', 'short'] 리스트를 Life is too short 문자열로 만들어 출력해 보자.**

In [0]:
a = ['Life', 'is', 'too', 'short']
b = " ".join(a)
print(b)

Life is too short


**Q8. (1,2,3) 튜플에 값 4를 추가하여 (1,2,3,4)를 만들어 출력해 보자.**

In [0]:
a = (1, 2, 3)
a = a + (4,)
print(a)

(1, 2, 3, 4)


**Q9. 다음과 같은 딕셔너리 a가 있다.**

a = dict()<br>
a<br>
{}<br>

다음 중 오류가 발생하는 경우를 고르고, 그 이유를 설명해 보자.

1. a['name'] = 'python'
2. a[('a',)] = 'python'
3. a[[1]] = 'python'
4. a[250] = 'python'
--------------------------------------------------------------------------------
각각 한번 실행 해 보자.

In [0]:
a = dict()
a

a['name'] = 'python'

오류가 발생하지 않는다.

In [0]:
a = dict()
a

a[('a',)] = 'python'

오류가 발생하지 않는다.

In [0]:
a = dict()
a

a[[1]] = 'python'

TypeError: unhashable type: 'list'

오류가 발생한다. 그 이유는 딕셔너리의 키는 변하는 값을 사용할 수 없기 때문이다. 위에서 사용한 [1]은 리스트로써 변하는 값이기 때문에 오류가 나타나는 것이다.

In [0]:
a = dict()
a

a[250] = 'python'

오류가 발생하지 않는다.

**Q10. 딕셔너리 a에서 'B'에 해당되는 값을 추출해 보자.**

In [0]:
a = {'A':90, 'B':80, 'c':70}
b = a.pop('B')
print(a)
print(b)

{'A': 90, 'c': 70}
80


**Q11. a 리스트에서 중복 숫자를 제거해 보자.**

In [0]:
a = [1, 1, 1, 2, 2, 3, 3, 3, 4, 4, 5]

aSet = set(a)
b = list(aSet)
print(b)

[1, 2, 3, 4, 5]


**리스트 자료형이 집합 자료형으로 변환되면서 중복된 값들은 사라지게 됨.**

**Q12.파이썬은 다음처럼 동일한 값에 여러 개의 변수를 선언할 수 있다. 다음과 같이 a, b 변수를 선언한 후 a의 두 번째 요솟값을 변경하면 b 값은 어떻게 될까? 그리고 이런 결과가 오는 이유에 대해 설명해 보자.**

In [0]:
a = b = [1, 2, 3]
a[1] = 4
print(b)

[1, 4, 3]


리스트 자료형으로 [1, 4, 3]이 출력된다. 왜냐하면 a 와 b 모두 동일한 [1, 2, 3]의 리스트 객체를 가리키고 있기 때문이다.

# 3장 연습문제

**Q1. 다음 코드의 결괏값은 무엇일까?**

In [0]:
a = "Life is too short, you need python"

if "wife" in a:
    print("wife")
elif "python" in a and "you" not in a:
    print("python")
elif "shirt" not in a:
    print("shirt")
elif "need" in a:
    print("need")
else:
    print("none")

shirt


3번째 조건과 4번째 조건이 참의 결과를 출력하고, 가장 먼저 참으로 나타난 조건은 3번째 이므로 "shirt" 가 출력된다.

**Q2. while문을 사용해 1부터 1000까지의 자연수 중 3의 배수의 합을 구해 보자.**

In [0]:
result = 0
i = 1
while i <= 1000:
    if i % 3 == 0:
        result += i
    i += 1
    
print(result)

166833


3의 배수는 3으로 나누었을 때 나누어 떨어지는(나머지가 0인)수이다. 이를 while문을 이용하여 반복해서 출력값을 구했더니 166833 이 출력되었다.

**Q3. while문을 사용하여 다음과 같이 별(*)을 표시하는 프로그램을 작성해 보자.**

\* <br> \*\* <br> \*\*\* <br> \*\*\*\* <br> \*\*\*\*\*


In [0]:
i = 0
while True:
    i += 1
    if i > 5: break
    print('*' * i)

*
**
***
****
*****


**Q4. for문을 사용해 1부터 100까지의 숫자를 출력해 보자.**

In [0]:
for i in range(1, 101):
    print(i)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100


**Q5. A 학급에 총 10명의 학생이 있다. 이 학생들의 중간고사 점수는 다음과 같다.**

[70, 60, 55, 75, 95, 90, 80, 80, 85, 100]

for문을 사용하여 A 학급의 평균 점수를 구해 보자.

In [0]:
A = [70,60,55,75,95,90,80,80,85,100]
total = 0

for score in A:
    total += score
    
average = total / len(A)
print(average)

79.0


**Q6. 리스트 중에서 홀수에만 2를 곱하여 저장하는 다음 코드가 있다.**

numbers = [1, 2, 3, 4, 5]
result = []
for n in numbers:
    if n % 2 == 1:
        result.append(n*2)
        
위 코드를 리스트 내포(list comprehension)를 사용하여 표현해 보자.

In [0]:
numbers = [1,2,3,4,5]
result = [n*2 for n in numbers if n%2==1]
print(result)

[2, 6, 10]
